homeassistant/components/hue/sensor.py

```diff
@@ -0,0 +1,324 @@
+    # changes to the second entity until the next polling update because the
+    # throttle decorator will prevent the call.
+
+    progress = None
+    sensor_progress = set()
+
+    async def request_update(object_id):
+        """Request an update.
+
+        We will only make 1 request to the server for updating at a time. If a
+        request is in progress, we will join the request that is in progress.
+
+        This approach is possible because should_poll=True. That means that
+        Home Assistant will ask sensors for updates during a polling cycle or
+        after it has called a service.
+
+        We keep track of the sensors that are waiting for the request to
+        finish. When new data comes in, we'll trigger an update for all
+        non-waiting sensors. This covers the case where a service is called to
+        enable 2 sensors but in the meanwhile some other sensor has changed
+        too.
+        """
+        nonlocal progress
+
+        sensor_progress.add(object_id)
+
+        if progress is not None:
+            return await progress
+
+        progress = asyncio.ensure_future(update_bridge())
+        result = await progress
+        progress = None
+        sensor_progress.clear()
+        return result
+
+    async def update_bridge():
+        """Update the values of the bridge.
+
+        Will update sensors from the bridge.
+        """
+        tasks = []
+        tasks.append(async_update_items(
+            hass, bridge, async_add_entities, request_update, cur_sensors,
+            sensor_progress
+        ))
+
+        await asyncio.wait(tasks)
+
+    await update_bridge()
+
+
+async def async_update_items(hass, bridge, async_add_entities,
+                             request_bridge_update, current, progress_waiting):
+    """Update sensors from the bridge."""
+    import aiohue
+
+    api = bridge.api.sensors
+
+    try:
+        start = monotonic()
+        with async_timeout.timeout(4):
+            await api.update()
+    except (asyncio.TimeoutError, aiohue.AiohueException) as err:
+        _LOGGER.debug('Failed to fetch sensor: %s', err)
+
+        if not bridge.available:
+            return
+
+        _LOGGER.error('Unable to reach bridge %s (%s)', bridge.host, err)
+        bridge.available = False
+
+        for sensor_id, sensor in current.items():
+            if sensor_id not in progress_waiting:
+                sensor.async_schedule_update_ha_state()
+
+        return
+
+    finally:
+        _LOGGER.debug('Finished sensor request in %.3f seconds',
+                      monotonic() - start)
+
+    if not bridge.available:
+        _LOGGER.info('Reconnected to bridge %s', bridge.host)
+        bridge.available = True
+
+    new_sensors = []
+    sensor_device_names = {}
+
+    for item_id in api:
+        if item_id not in current:
+            name = PRESENCE_NAME_FORMAT.format(api[item_id].name)
+            if api[item_id].type == aiohue.sensors.TYPE_ZLL_PRESENCE:
```

Comment:
Why can it only ever happen that there are new presence sensors found and not temperature etc?

Reply:
And don't just comment here, but put the actual comment in the code.